src/backend/commands/variable.c

```diff
@@ -527,6 +527,17 @@ check_XactIsoLevel(int *newval, void **extra, GucSource source)
 {
 	int			newXactIsoLevel = *newval;
 
+	/*
+	 * FIXME: we cannot support XACT_REPEATABLE_READ now due to concurrent 
+	 * insert. In concurrent insert, one transaction apply new change based
+	 * on the result of other transaction done. Thus, XACT_REPEATABLE_READ
+	 * is not support and will be support soon.
+	 */
+	if(enable_serverless && newXactIsoLevel == XACT_REPEATABLE_READ){
```

Comment:
Is it possible we disable it in extension instead of here?

Reply:
Yes, it's a better idea to move it into extension. And where and when is an intersting problem.